lib/cc/yaml/nodes/root.rb

```diff
@@ -16,6 +20,17 @@ def nested_warnings(*)
         def inspect
           "#<#{self.class.name}:#{super}>"
         end
+
+        def exclude?(path)
+          exclude_paths.present? &&
+            exclude_paths.match_any?(path)
+        end
+
+        def rate?(path)
+          ratings.nil? ||
+            ratings.paths.blank? ||
```

Comment:
Agreed, but I'd like to keep a delegator here too to encapsulate and
centralize the missing key portion of the logic.

On Sun, Jun 14, 2015, 14:29 Bryan Helmkamp <notifications@github.com> wrote:

> In lib/cc/yaml/nodes/root.rb
> <https://github.com/codeclimate/codeclimate-yaml/pull/1#discussion_r32382671>
> :
>
> > @@ -16,6 +20,17 @@ def nested_warnings(*)
> >          def inspect
> >            "#<#{self.class.name}:#{super}>"
> >          end
> > +
> > +        def exclude?(path)
> > +          exclude_paths.present? &&
> > +            exclude_paths.match_any?(path)
> > +        end
> > +
> > +        def rate?(path)
> > +          ratings.nil? ||
> > +            ratings.paths.blank? ||
>
> Should this method live on Ratings?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codeclimate/codeclimate-yaml/pull/1/files#r32382671>.
>